### PR TITLE
Slack High PID - solution

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22018

## Description
Implementing @GalRabin solution 
```python
        from concurrent.futures import ThreadPoolExecutor
        with ThreadPoolExecutor() as p:
            loop = asyncio.get_event_loop()
            loop.set_default_executor(p)
            command_func()
```
as mentioned here: https://github.com/demisto/etc/issues/22018#issuecomment-610493091

Tested all the commands with this solution and it results with only 1 active PID on any active container the server creates.

### CODE REVIEW

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import demistomock as demisto
 from CommonServerPython import *
-from CommonServerUserPython import *
+#from CommonServerUserPython import *
```
Comment (ronykoz) [inline Packs/Slack/Integrations/Slack/Slack.py]: remove the comment

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 2]
```diff
@@ -10,7 +10,12 @@
 import concurrent
 import requests
 import ssl
+import json
+import re
+import time
+from datetime import datetime
```
Comment (ronykoz) [inline Packs/Slack/Integrations/Slack/Slack.py:16]: Seems that you've added a bunch of not used imports, please sort it out

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 3]
```diff
@@ -1834,7 +1841,7 @@ def init_globals():
     proxy_url = demisto.params().get('proxy_url')
     PROXY_URL = proxy_url or PROXIES.get('http')  # aiohttp only supports http proxy
     DEDICATED_CHANNEL = demisto.params().get('incidentNotificationChannel')
-    CLIENT = slack.WebClient(token=BOT_TOKEN, proxy=PROXY_URL, ssl=SSL_CONTEXT)
+    CLIENT = slack.WebClient(token=BOT_TOKEN, proxy=PROXY_URL, ssl=SSL_CONTEXT, )
```
Comment (ronykoz) [inline Packs/Slack/Integrations/Slack/Slack.py:1845]: undo

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 4]
```diff
@@ -1873,8 +1901,16 @@ def main():
...
+        # Run non-long running commands
+        if command_func != long_running_main:
+            timed_thread = SlackCommandTimedThread(command_func, SLACK_COMMAND_TIMEOUT)
+            timed_thread.timed_run()
+
+        else:
+            long_running_main()
+
     except Exception as e:
+        print(e)
```
Comment (ronykoz) [inline Packs/Slack/Integrations/Slack/Slack.py:1915]: remove

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 5]
```diff
@@ -1844,6 +1852,28 @@ def init_globals():
...
+
+    def timed_run(self):
+        """Run the thread for the given time.
+
+        Returns:
+
+        """
+        self.start()
+        time.sleep(self.timeout)
+        self.join()
```
Comment (ronykoz) [inline Packs/Slack/Integrations/Slack/Slack.py:1876]: where are you exiting the thread?

File: Packs/Slack/Integrations/Slack/Slack.py (hunk 2 above)
Comment (roysagi) [reply Packs/Slack/Integrations/Slack/Slack.py:16]: done - just added this cause it raised warnings within the code.

File: Packs/Slack/Integrations/Slack/Slack.py (hunk 3 above)
Comment (roysagi) [reply Packs/Slack/Integrations/Slack/Slack.py:1845]: done

File: Packs/Slack/Integrations/Slack/Slack.py (hunk 5 above)
Comment (roysagi) [reply Packs/Slack/Integrations/Slack/Slack.py:1876]: no need apperantly - I'll update the description

File: Packs/Slack/Integrations/Slack/Slack.py (hunk 4 above)
Comment (roysagi) [reply Packs/Slack/Integrations/Slack/Slack.py:1915]: done

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 6]
```diff
@@ -26,7 +27,7 @@
     'Critical': 4
 }
 
-
+LOOP = None
```
Comment (ronykoz) [inline Packs/Slack/Integrations/Slack/Slack.py:30]: remove this one too

Comment (roysagi) [reply Packs/Slack/Integrations/Slack/Slack.py]: done

**Comment by glicht** [comment]
opened a new pr: https://github.com/demisto/content/pull/6411

**Comment by roysagi** [comment]
closing as @glicht requested - issue should be solved by: https://github.com/demisto/content/pull/6411

**Pull request closed without merging**